### PR TITLE
Add support for tracking checklist progress

### PR DESCRIPTION
**Pull request by aaronpoweruser**
- Adds a new setting for a reference note to compare.
- Marks any checked task as complete for the day.

### CODE REVIEW

**Comment by jgclark** [comment]
Firstly, well done for understanding the quite complex code well enough to make any addition. It has evolved significantly over the years, and so the design may rightly feel a bit odd. Secondly, I hope this means my commenting was in pretty good shape.
Thirdly, this feels sufficiently different from the other ways that summaries work that it will definitely need a new subsection in the README before it's released. I think a before and after example would be good.

**Comment by aaronpoweruser** [comment]
@jgclark updated the documentation, any other changes?

File: jgclark.Summaries/src/todayProgress.js [hunk 1]
```diff
@@ -132,6 +132,7 @@ export async function makeTodayProgress(itemsToShowArr: Array<string> = [], sour
       GOMentionsTotal: (mentionsToShow.length > 0) ? mentionsToShow : [],
       GOMentionsAverage: [],
       GOMentionsExclude: [],
+      GOChecklistRefNote: string,
```
Comment (jgclark) [inline jgclark.Summaries/src/todayProgress.js:135]: This is an error. You probably mean ''.

File: jgclark.Summaries/src/progress.js [hunk 2]
```diff
@@ -128,6 +128,7 @@ export async function makeProgressUpdate(paramsIn: any = '', source: string = 'c
         GOMentionsTotal: paramProgressMentionsTotal,
         GOMentionsAverage: paramProgressMentionsAverage,
         GOMentionsExclude: [],
+        GOChecklistRefNote: config.progressChecklistReferenceNote,
```
Comment (jgclark) [inline jgclark.Summaries/src/progress.js:131]: This doesn't look ideal, as it breaks the clear pattern above. 
What you want is to add the ability to pass this in as a parameter, like all the others, unless there's a good reason it doesn't make sense.

**Review by jgclark — CHANGES_REQUESTED** [review]
Please see just 2 lines where I think changes are needed.
I've made a few typo corrections in the README.

**Comment by jgclark** [comment]
Well done again, @aaronpoweruser. 
Good news:  I've asked Eduard to give you write permission for this repo, and he says he's now done so.
So please enjoy being able to merge and then release this yourself. 
To release use `npc plugin:release jgclark.Dashboard` 
and then add a comment in the Discord channel that's similar to the CHANGELOG entry.

**Pull request merged**